common/src/main/java/com/microsoft/identity/common/internal/cache/BrokerOAuth2TokenCache.java

```diff
@@ -198,7 +198,8 @@ public ICacheRecord save(@NonNull AccountRecord accountRecord,
             // Save to the processUid cache... or create a new one
```

Comment:
I think we shouldn't be even saving access and id token records to foci cache, right? How will id and access tokens be saved to uid based cache if it's a foci token?

Reply:
Apps will use _either_ the foci cache or the uid cache, but not both.

If an app is not FoCI:
- The `uid`-based cache is used to persist all tokens (+accounts)

If an app is FoCI:
- The `foci` cache is used to store all tokens (+accounts)
- When the presented `uid` matches the `BrokerApplicationMetadata` all tokens (for the requesting account) are returned
- When the presented `uid` _doesn't_ match, only the RT and Account are returned
    * The controller above will then see that no AT was returned and make a `/token` request
    * If the request is successful, the controller will call `save()` and a `BrokerApplicationMetadata` entry will be created
    * If the request is not successful, an the error is wrapped in an Exception and returned to the client